Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1019,10 +1019,7 @@ bool RenderFlexibleBox::childCrossSizeShouldUseContainerCrossSize(const RenderBo
     if (!isMultiline() && alignmentForChild(child) == ItemPosition::Stretch && !hasAutoMarginsInCrossAxis(child) && crossSizeLengthForChild(MainOrPreferredSize, child).isAuto()) {
         if (crossAxisIsPhysicalWidth())
             return true;
-        // This must be kept in sync with computeMainSizeFromAspectRatioUsing().
-        // FIXME: so far we're only considered fixed sizes but we should extend it to other definite sizes.
-        auto& crossSize = isHorizontalFlow() ? style().height() : style().width();
-        return crossSize.isFixed();
+        return hasDefiniteLogicalHeight(); 
```

Comment:
_hasDefiniteLogicalHeight_ calls _availableLogicalHeightForPercentageComputation_ which has this odd early return on _skipContainingBlockForPercentHeightCalculation_.  It makes me a bit uneasy and it may return false for fixed width box. We should probably go back to the original approach here.